### PR TITLE
Port CUDA implementation of expm1 to ATen

### LINKED ISSUE #24562: Migrate `expm1` and `expm1_` from the TH to Aten (CUDA)
Porting TH operators is essential for code simplicity and performance reasons.

Porting guides and Q&A are available in umbrella issue: #24507

Feel free to add @VitalyFedyunin as a reviewer to get a prioritized review.

### PR DESCRIPTION
**Pull request by vishwakftw**
Closes #24562

### CODE REVIEW

File: aten/src/ATen/native/cuda/UnaryOpsKernel.cu [hunk 1]
```diff
@@ -40,6 +40,15 @@ void ceil_kernel_cuda(TensorIterator& iter) {
   });
 }
 
+void expm1_kernel_cuda(TensorIterator& iter) {
+  AT_DISPATCH_FLOATING_TYPES_AND_HALF(iter.dtype(), "expm1_cuda", [&]() {
+    gpu_kernel(iter, []GPU_LAMBDA(scalar_t a) -> scalar_t {
+      return ::expm1(a);
```
Comment (xuhdev) [inline aten/src/ATen/native/cuda/UnaryOpsKernel.cu]: Could you verify that this is overloaded on CUDA?

Comment (vishwakftw) [reply aten/src/ATen/native/cuda/UnaryOpsKernel.cu]: Yes, quoting the CUDA [Math API reference](https://docs.nvidia.com/cuda/cuda-math-api/group__CUDA__MATH.html#group__CUDA__MATH): "Note that floating-point functions are overloaded for different argument types"

**Review by xuhdev — APPROVED** [review]

**Review by VitalyFedyunin — APPROVED** [review]
Please rebase

**Comment by vishwakftw** [comment]
@VitalyFedyunin I’ve rebased.

**Comment by VitalyFedyunin** [comment]
Sorry `UnaryOps.cpp` have high contention, you need to rebase again.

**Comment by vishwakftw** [comment]
@VitalyFedyunin no worries, I've rebased again.

**Comment by VitalyFedyunin** [comment]
Fails to build

**Comment by vishwakftw** [comment]
@VitalyFedyunin I've fixed the rebase, sorry about earlier.

**Comment by vishwakftw** [comment]
@VitalyFedyunin is this good to go?

**Pull request closed without merging**